app/src/main/java/org/openobservatory/ooniprobe/activity/ResultDetailActivity.java

```diff
@@ -105,6 +110,54 @@ protected void onResume() {
         load();
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.rerun, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onPrepareOptionsMenu(Menu menu) {
+        invalidateOptionsMenu();
+        if (!result.test_group_name.equals(WebsitesSuite.NAME))
+            menu.findItem(R.id.reRun).setVisible(false);
+        return super.onPrepareOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.reRun:
+                new ConfirmDialogFragment.Builder()
+                        .withExtra("re_run")
+                        //TODO sub these rows when the string is inserted
+                        //.withMessage(getString(R.string.Modal_ReRun_Websites_Title))
+                        //.withPositiveButton(getString(R.string.Modal_ReRun_Websites_Run))
+                        .withMessage(getString(R.string.Modal_DoYouWantToDeleteThisTest))
+                        .withPositiveButton(getString(R.string.Modal_Delete))
+                        .build().show(getSupportFragmentManager(), null);
```

Comment:
I am missing why we're offering to delete a test when the user clicks on rerun

Reply:
Placeholder strings. Read above answer